qiskit_nature/problems/second_quantization/vibrational/spin_op_builder.py

```diff
@@ -0,0 +1,68 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+""" Spin operator builder. """
+
+from qiskit_nature import WatsonHamiltonian
+from qiskit_nature.components.bosonic_bases import HarmonicBasis
+
+
+# TODO complete once a relevant Vibrational SpinOp class created
+def build_spin_op(watson_hamiltonian: WatsonHamiltonian, basis_size, truncation_order):
+    """
+    Builds a spin operator based on a WatsonHamiltonian object.
+
+    Args:
+        watson_hamiltonian (WatsonHamiltonian): WatsonHamiltonian instance.
+        basis_size: size of a basis
+        truncation_order: order at which an n-body expansion is truncated
+
+    Returns:
+        SpinOp: SpinOp built from a QMolecule object.
+    """
+
+    num_modes = watson_hamiltonian.num_modes
+    basis_size = [basis_size] * num_modes
+    boson_hamilt_harm_basis = HarmonicBasis(watson_hamiltonian,  # type: ignore
```

Comment:
I think we should make `HarmonicBasis` configurable and instead take any `BosonicBasis` as an input somewhere.

Reply:
Sure. Let's deal with it a bit later. Currently we only have a `HarmonicBasis` implemented.